### PR TITLE
Fixed #25009 - User.objects.create_user(...,is_staff=True) raises TypeError

### PR DESCRIPTION
**Pull request by pahko**
https://code.djangoproject.com/ticket/25009

### CODE REVIEW

File: django/contrib/auth/models.py [hunk 1]
```diff
@@ -132,28 +132,37 @@ def natural_key(self):
...
 
     def create_superuser(self, username, email, password, **extra_fields):
-        return self._create_user(username, email, password, True, True,
-                                 **extra_fields)
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+
+        if not extra_fields.get('is_staff'):
+            raise ValueError(
+                'Superuser needs to be staff. Do not set is_staff.'
```
Comment (coldmind) [inline django/contrib/auth/models.py]: `Do not set is_staff.` sounds tricky.
`is_staff must be set to True` sounds more clearly.

File: django/contrib/auth/models.py [hunk 2]
```diff
@@ -132,28 +132,37 @@ def natural_key(self):
...
+        extra_fields.setdefault('is_superuser', False)
+        return self._create_user(username, email, password, **extra_fields)
 
     def create_superuser(self, username, email, password, **extra_fields):
-        return self._create_user(username, email, password, True, True,
-                                 **extra_fields)
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+
+        if not extra_fields.get('is_staff'):
```
Comment (coldmind) [inline django/contrib/auth/models.py]: IMO `if extra_fields.get('is_staff') is not True:` represents what need to be checked here more clearly.

Comment (pahko) [inline django/contrib/auth/models.py]: @coldmind  in this case `is not True` is not what we want to test, what we want to test is that the value of`is_staff` evaluated to boolean is False; `1` is a valid case that `is not True` does not catch.

I updated the message :)

Comment (coldmind) [inline django/contrib/auth/models.py]: @pahko `2` , `3` and non-empty lists and objects also will be valid case. Only check for boolean is needed here.

Comment (pahko) [inline django/contrib/auth/models.py]: @coldmind I see your point... explicit `True|False`... pr updated! thanks!!

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
merged in e75b614640c206bf6d0b1c9d32c54434ea719582, thanks!

**Pull request closed without merging**